optaplanner-core/src/main/java/org/optaplanner/core/api/score/buildin/simple/SimpleScore.java

```diff
@@ -169,15 +170,11 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-        // A direct implementation (instead of HashCodeBuilder) to avoid dependencies
-        return ((17 * 37)
-                + initScore) * 37
-                + score;
+        return Objects.hash(initScore, score);
     }
 
     @Override
     public int compareTo(SimpleScore other) {
-        // A direct implementation (instead of CompareToBuilder) to avoid dependencies
         if (initScore != other.getInitScore()) {
             return initScore < other.getInitScore() ? -1 : 1;
```

Comment:
Nitpick: we can use return Integer.compare(initScore, other.getInitScore()); here too